mds/management/commands/poll_providers.py

```diff
@@ -199,15 +200,27 @@ def get_body(self, provider, url):
     def process_status_changes(self, status_changes, provider):
         if self.verbosity > 1:
             self.stdout.write("Processing...")
+
+        # What is the latest event recorded in that series? (order is not asserted in the specs)
+        try:
+            last_start_time_polled = utils.from_mds_timestamp(
+                max(status_change["event_time"] for status_change in status_changes)
+            )
+        except (KeyError, ValueError):
+            # Data so bad there is no or nothing but invalid event times
```

Comment:
actually you break whenever there is a corrupt event, no ?

Reply:
There is another function before to validate data.